Dockerfiles/agent-ddot/Dockerfile.agent-otel

```diff
@@ -1,14 +1,12 @@
-ARG AGENT_REPO=datadog/agent-dev
-ARG AGENT_VERSION=nightly-full-main-jmx
-ARG AGENT_BRANCH=main
+ARG AGENT_DOCKER_REPO=datadog/agent
```

Comment:
We had agreed in the past that the `main` BYOC dockerfile would build a dev nightly image, hence why it points to the development image. If you look at our stable release branches, you will see that these have been modified by an invoke task: 
```
dda inv -- -e otel-agent.byoc-release -b 7.71.2 -i 7.71.2 -r datadog/agent
```
For example. 

The problem, and I see you try to solve that here is that `AGENT_VERSION` and `AGENT_DOCKER_TAG` could be mixed and matched. This looks like we're on the right track, but we probably need to fix the invoke task as well.

I realize that we might have a "problem" if we try to support the nightly builds if we want to prevent the mix and match problem 🤔